python/nav/web/api/v1/views.py

```diff
@@ -1153,3 +1157,105 @@ class ModuleViewSet(NAVAPIMixin, viewsets.ReadOnlyModelViewSet):
         'device__serial',
     )
     serializer_class = serializers.ModuleSerializer
+
+
+class VendorLookup(NAVAPIMixin, APIView):
+    """Lookup vendor names for MAC addresses.
+
+    This endpoint allows you to look up vendor names for MAC addresses.
+    It can be used with either a GET or POST request.
+
+    For GET requests, the MAC address must be provided via the query parameter `mac`.
+    This only supports one MAC address at a time.
+
+    For POST requests, the MAC addresses must be provided in the request body
+    as a JSON array. This supports multiple MAC addresses.
+
+    Responds with a JSON dict mapping the MAC addresses to the corresponding vendors.
+    The MAC addresses will have the format `aa:bb:cc:dd:ee:ff`. If the vendor for a
+    given MAC address is not found, it will be omitted from the response.
+
+    Example GET request: `/api/1/vendor/?mac=aa:bb:cc:dd:ee:ff`
+
+    Example GET response: `{"aa:bb:cc:dd:ee:ff": "Vendor A"}`
+
+    Example POST request:
+        `/api/1/vendor/` with body `["aa:bb:cc:dd:ee:ff", "11:22:33:44:55:66"]`
+
+    Example POST response:
+        `{"aa:bb:cc:dd:ee:ff": "Vendor A", "11:22:33:44:55:66": "Vendor B"}`
+    """
+
+    @staticmethod
+    def get(request):
+        mac = request.GET.get('mac', None)
+        if not mac:
+            return Response("Missing MAC address", status=status.HTTP_400_BAD_REQUEST)
+        try:
+            results = get_vendor_names([mac])
+        except ValueError:
+            return Response("Invalid MAC address", status=status.HTTP_400_BAD_REQUEST)
+        return Response(results)
+
+    @staticmethod
+    def post(request):
+        try:
+            mac_addresses = json.loads(request.body.decode('utf-8'))
+        except json.JSONDecodeError:
+            return Response("Invalid JSON", status=status.HTTP_400_BAD_REQUEST)
+
+        if not isinstance(mac_addresses, list):
+            return Response(
+                "JSON body must represent a list of MAC addresses",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        try:
+            results = get_vendor_names(mac_addresses)
+        except ValueError:
+            return Response("Invalid MAC address", status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Would be nice to here return which one(s) is/are invalid

Reply:
This has been added